src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -44,6 +45,15 @@ public function process(ContainerBuilder $container)
                     $this->completeDefinition($id, $definition);
                 }
             }
+
+            foreach ($this->usedTypes as $type => $id) {
+                if (isset($this->usedTypes[$type]) && isset($this->notGuessableTypes[$type])) {
```

Comment:
That was a good question actually, that made me catch a bug (see https://github.com/symfony/symfony/pull/21670).